### PR TITLE
Create LBFGSSolver structure

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by dpo** [comment]
Looks great! Is it possible to add an allocation test? The LBFGS operator itself can probably be improved.

**Comment by abelsiqueira** [comment]
Maybe after updating the LBFGS Operator.

**Comment by abelsiqueira** [comment]
Will the LO revamp change LBFGS a lot? Should I wait?

**Comment by abelsiqueira** [comment]
profile
Profile comparing the new version `c...` with the old `8...`. Problem with up to 1000 variables.

**Comment by abelsiqueira** [comment]
Even after updating the LinearOperator LBFGS, I couldn't find a clear way to measure the allocations. I even tried using a DummyModel with 0 allocation, but still there are too many moving parts, with the `armijo_wolfe` search and `LineModel`.

File: src/lbfgs.jl [hunk 1]
```diff
@@ -49,9 +80,10 @@ function lbfgs(
   status = :unknown
 
   h = LineModel(nlp, x, ∇f)
```
Comment (tmigot) [inline src/lbfgs.jl:82]: Shouldn't the `LineModel` also be allocated in the `LBFGSSolver` ?

**Comment by dpo** [comment]
Also, have a look at the LinearOperators revamp.

**Comment by abelsiqueira** [comment]
Now working with the least allocation possible, which is the ExecutationStats.
Should I try adding these to the solver as well? It involves more changes to SolverCore, probably.

**Review by dpo — APPROVED** [review]

**Pull request merged**